lib/subjects/seen_remover.rb

```diff
@@ -0,0 +1,36 @@
+module Subjects
+  class SeenRemover
+
+    attr_reader :user, :workflow, :append_sms_ids
+
+    def initialize(user, workflow, append_ids)
+      @user = user
+      @workflow = workflow
+      @append_sms_ids = append_ids
```

Comment:
This one is a very subtle point, so don't hesitate to disagree: to me it feels more like `user` and `workflow` make up the "context" for this operation, whereas `append_ids` is the main piece of data. Because of that, I would have personally made `append_ids` a parameter to `unseen_ids` (and passing it along as a param to `seen_before_sms_ids`). The downside is having to pass that along, the upside is that it makes it easier to "reuse". Not sure if we'll do that in production code, but for instance in the tests you could use the same object but call `unseen_ids` with different arguments: an empty array vs. a normal one comes to mind.